pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
+
+            if isinstance(ctx.gradient_fn, qml.gradients.gradient_transform):
+                # Gradient function is a gradient transform.
+
+                # Generate and execute the required gradient tapes
+                if ctx._n < ctx.max_diff:
+                    # The derivative order is less than the max derivative order.
+                    # Compute the VJP recursively by using the gradient transform
+                    # and calling ``execute`` to compute the results.
+                    # This will allow higher-order derivatives to be computed
+                    # if requested.
+
+                    vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                        ctx.tapes,
+                        dy,
+                        ctx.gradient_fn,
+                        reduction="extend",
+                        gradient_kwargs=ctx.gradient_kwargs,
+                    )
+                    # This is where the magic happens. Note that we call ``execute``.
+                    # This recursion, coupled with the fact that the gradient transforms
+                    # are differentiable, allows for arbitrary order differentiation.
+                    res = execute(
+                        vjp_tapes,
+                        ctx.device,
+                        ctx.execute_fn,
+                        ctx.gradient_fn,
+                        ctx.gradient_kwargs,
+                        _n=ctx._n + 1,
+                        max_diff=ctx.max_diff,
+                    )
+                    vjps = processing_fn(res)
+
+                else:
+                    # The derivative order is at the maximum. Compute the VJP
+                    # in a non-differentiable manner to reduce overhead.
+                    with qml.tape.Unwrap(*ctx.tapes):
+                        vjp_tapes, processing_fn = qml.gradients.batch_vjp(
+                            ctx.tapes,
+                            dy,
+                            ctx.gradient_fn,
+                            reduction="extend",
+                            gradient_kwargs=ctx.gradient_kwargs,
+                        )
+
+                        vjps = processing_fn(ctx.execute_fn(vjp_tapes)[0])
+
+            else:
+                # Gradient function is not a gradient transform
+                # (e.g., it might be a device method).
+                # Note that unlike the previous branch:
+                #
+                # - there is no recursion here
+                # - gradient_fn is not differentiable
+                #
+                # so we cannot support higher-order derivatives.
+
+                with qml.tape.Unwrap(*ctx.tapes):
+                    jacs = ctx.gradient_fn(ctx.tapes, **ctx.gradient_kwargs)
+
+                vjps = _compute_vjps_new(dy, jacs, multi_measurements, device=ctx.torch_device)
+
+        # Remove empty vjps
+        vjps = [vjp for vjp in vjps if not isinstance(vjp, np.ndarray)]

```

Comment:
Why are there empty vjps?

Reply:
Because it is possible for some tangents to be all zeros, in that case we return an empty jacobian, but torch does not appreciate it contrarily to other frameworks.